### PR TITLE
Fix up event declaration

### PR DESCRIPTION
**Pull request by benrudolph**
@biyeun fixes up some stuff with the messaging api. the django scheme was returning `http` for some reason which didn't match the `https` scheme so it broke the origin check. now only checks for the proper hostname. also fixed the click event.

cc: @emord @nickpell

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js [hunk 1]
```diff
@@ -186,3 +186,15 @@ if (!String.prototype.startsWith) {
         return this.substr(position, searchString.length) === searchString;
     };
 }
+
+if (!String.prototype.endsWith) {
+  String.prototype.endsWith = function(searchString, position) {
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js]: Expected indentation of 4 space characters but found 2. (indent)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js [hunk 2]
```diff
@@ -186,3 +186,15 @@ if (!String.prototype.startsWith) {
         return this.substr(position, searchString.length) === searchString;
     };
 }
+
+if (!String.prototype.endsWith) {
+    String.prototype.endsWith = function(searchString, position) {
```
Comment (emord) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js]: Is this for phantomjs, or does cloudcare target old browsers?

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js [hunk 3]
```diff
@@ -200,18 +200,18 @@ FormplayerFrontend.on("start", function (options) {
...
+            new FormplayerFrontend.HQ.Events.Receiver(options.allowedHost),
             false
         );
     }
 });
 
 FormplayerFrontend.on('navigation:back', function() {
     var url = Backbone.history.getFragment();
-    if (!url.startsWith('/single_app')) {
+    if (url.indexOf('single_app') === -1) {
```
Comment (emord) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js]: could use `includes` to be slightly clearer? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

**Review by emord — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js (hunk 3 above)
Comment (emord) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js]: You could also use a polyfill (on mdn from that link)

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 203 - Expected indentation of 4 space characters but found 2. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 204 - Expected indentation of 6 space characters but found 4. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 205 - Expected indentation of 6 space characters but found 4. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 206 - Expected indentation of 8 space characters but found 6. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 208 - Expected indentation of 6 space characters but found 4. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 209 - Expected indentation of 8 space characters but found 6. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 211 - Expected indentation of 8 space characters but found 6. (indent)

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 203 - Expected indentation of 4 space characters but found 2. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 204 - Expected indentation of 6 space characters but found 4. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 205 - Expected indentation of 6 space characters but found 4. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 206 - Expected indentation of 8 space characters but found 6. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 208 - Expected indentation of 6 space characters but found 4. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 209 - Expected indentation of 8 space characters but found 6. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/util.js, line 211 - Expected indentation of 8 space characters but found 6. (indent)

**Comment by benrudolph** [comment]
actually think dimagimon might be confused in this case. ran my local beautifier and it doesn't seem like dimagimon is counting the spaces correctly? maybe i'm going crazy

**Pull request merged**